src/redux/accessoriesRedux.tsx

```diff
@@ -1,19 +1,19 @@
 import { API_URL } from "../config";
+import { Store } from "./initialState";
+import { ProductType } from "../types/ProductType";
+import { CartItem } from "../types/CartItem";
+
 
 //selectors
-export const getGiftAccessoryByKickscooter = ({ accessories }, gift) =>
+export const getGiftAccessoryByKickscooter = ({ accessories }: Store, gift: string) =>
   accessories.find((accessory) => accessory.title === gift);
-export const getAllAccessories = (state) => state.accessories;
+export const getAllAccessories = (state: Store) => state.accessories;
 
-// action name creator
-const createActionName = (actionName) => `app/accessories/${actionName}`;
 
-// action types
-const LOAD_ACCESSORIES = createActionName("LOAD_ACCESSORIES");
 
 // action creators
-export const loadAccessories = (payload) => ({
-  type: LOAD_ACCESSORIES,
+export const loadAccessories = (payload: ProductType) => ({
+  type: "LOAD_ACCESSORIES",
```

Comment:
Typescript will actually infer the return type of this function as `{ type: string, payload: ProductType }` when we in fact need it to be `{ type: "LOAD_ACCESSORIES", payload: ProductType }`. You can make it do so by either adding `"LOAD_ACCESSORIES" as const` or by annotating return type as `(payload: ProductType): LoadAccessoriesAction => {...`